File: src/main/java/com/laamella/code_state_machine/condition/NonEventBasedCondition.java
package com.laamella.code_state_machine.condition;

import com.laamella.code_state_machine.Condition;

/**
 * A base class for conditions that do not respond to events.
 */
public abstract class NonEventBasedCondition<E> implements Condition<E> {
	@Override
	public final void handleEvent(final E event) {
		// Not event based, so not used.
	}

	@Override
	public void reset() {
		// Does nothing by default.
	}
}

File: scala/src/main/scala/com/laamella/code_state_machine/condition/NonEventBasedCondition.scala
package com.laamella.code_state_machine.condition

import com.laamella.code_state_machine.Condition

/**
 * A base class for conditions that do not respond to events.
 */
abstract class NonEventBasedCondition[E] extends Condition[E] {
	// Not event based, so not used.
	override def handleEvent(event: E): Unit = Unit

	// Does nothing by default.
	override def reset(): Unit = Unit
}
